components/file_preview/file_preview.tsx

```diff
@@ -1,38 +1,32 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, {ReactNode} from 'react';
 import {getFileThumbnailUrl, getFileUrl} from 'mattermost-redux/utils/file_utils';
 
 import FilenameOverlay from 'components/file_attachment/filename_overlay.jsx';
 import RemoveIcon from 'components/widgets/icons/fa_remove_icon';
 import Constants, {FileTypes} from 'utils/constants';
 import * as Utils from 'utils/utils.jsx';
 
-import FileProgressPreview from './file_progress_preview.jsx';
+import FileProgressPreview from './file_progress_preview';
 
-export default class FilePreview extends React.PureComponent {
-    static propTypes = {
-        enableSVGs: PropTypes.bool.isRequired,
-        onRemove: PropTypes.func.isRequired,
-        fileInfos: PropTypes.arrayOf(PropTypes.object).isRequired,
-        uploadsInProgress: PropTypes.array,
-        uploadsProgressPercent: PropTypes.object,
-    };
-
-    static defaultProps = {
-        fileInfos: [],
-        uploadsInProgress: [],
-        uploadsProgressPercent: {},
-    };
-
-    handleRemove = (id) => {
+type Props = {
+    enableSVGs: boolean;
+    onRemove: (id: string) => void;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    fileInfos: any[];
+    uploadsInProgress: string[];
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    uploadsProgressPercent: Map<string, any>;
```

Comment:
Same with this type, which should also be `FilePreviewInfo`

Reply:
These two props were optional, and now are required. Is this intentional?

Also we've changed the Props contract on the datatype here. Is the parent supposed to pass in an object or a Map? For an object you can use this type:

```ts
uploadsProgressPercent: {[clientID: string]: FilePreviewInfo}
```